dds_web/web/user.py

```diff
@@ -173,6 +176,10 @@ def login():
     if not dds_web.utils.is_safe_url(next):
         return flask.abort(400)
 
+    # Display greeting message, if applicable
+    if re.search("confirm_deletion", next):
```

Comment:
Change to
```
    if next and re.search("confirm_deletion", next):
```
Otherwise this results in an error. 
To reproduce: 
1. `dds user delete --self`
2. Open email and go to confirm link - should result in the success message, the user's removed
3. Edit the url to only `127.0.0.1:5000` --> goes to login page, but results in:
```
dds_backend | 172.19.0.1 - - [24/Jan/2022 13:46:06] "GET /login HTTP/1.1" 500 -
dds_backend | Traceback (most recent call last):
dds_backend |   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 2464, in __call__
dds_backend |     return self.wsgi_app(environ, start_response)
dds_backend |   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 2450, in wsgi_app
dds_backend |     response = self.handle_exception(e)
dds_backend |   File "/usr/local/lib/python3.10/site-packages/flask_restful/__init__.py", line 272, in error_router
dds_backend |     return original_handler(e)
dds_backend |   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 1867, in handle_exception
dds_backend |     reraise(exc_type, exc_value, tb)
dds_backend |   File "/usr/local/lib/python3.10/site-packages/flask/_compat.py", line 39, in reraise
dds_backend |     raise value
dds_backend |   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 2447, in wsgi_app
dds_backend |     response = self.full_dispatch_request()
dds_backend |   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 1952, in full_dispatch_request
dds_backend |     rv = self.handle_user_exception(e)
dds_backend |   File "/usr/local/lib/python3.10/site-packages/flask_restful/__init__.py", line 272, in error_router
dds_backend |     return original_handler(e)
dds_backend |   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 1821, in handle_user_exception
dds_backend |     reraise(exc_type, exc_value, tb)
dds_backend |   File "/usr/local/lib/python3.10/site-packages/flask/_compat.py", line 39, in reraise
dds_backend |     raise value
dds_backend |   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 1950, in full_dispatch_request
dds_backend |     rv = self.dispatch_request()
dds_backend |   File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 1936, in dispatch_request
dds_backend |     return self.view_functions[rule.endpoint](**req.view_args)
dds_backend |   File "/usr/local/lib/python3.10/site-packages/flask_limiter/extension.py", line 702, in __inner
dds_backend |     return obj(*a, **k)
dds_backend |   File "/code/dds_web/web/user.py", line 181, in login
dds_backend |     if re.search("confirm_deletion", next):
dds_backend |   File "/usr/local/lib/python3.10/re.py", line 200, in search
dds_backend |     return _compile(pattern, flags).search(string)
dds_backend | TypeError: expected string or bytes-like object
```
Since `next` is not set

Reply:
[Ok, found it.](https://stackoverflow.com/a/16069560)

Reply:
Yes! As you've found, if the first part of the if statement fails then it doesn't continue.